include/drivers/gpio.h

```diff
@@ -304,6 +304,59 @@ typedef uint8_t gpio_dt_flags_t;
  */
 typedef uint32_t gpio_flags_t;
 
+/**
+ * @brief Provides a type to hold GPIO information specified in devicetree
+ *
+ * This type is sufficient to hold a GPIO device pointer, pin number,
+ * and the subset of the flags used to control GPIO configuration
+ * which may be given in devicetree.
+ */
+struct gpio_dt_spec {
+	const struct device *gpio;
```

Comment:
Nit: gpio_device/gpio_dev/dev_gpio ?

Reply:
Or `ctlr` which might match the devicetree nomenclature better.

Reply:
> Nit: gpio_device/gpio_dev/dev_gpio ?

I like a nice short name. I think I'll go with `port` for consistency with what the rest of include/drivers/gpio.h uses.

Reply:
> Or `ctlr` which might match the devicetree nomenclature better.

Saw this too late, but I still think `port` is clearer since it matches the rest of the API calls. We should fix the impedance mismatch between 'controller' and 'port' at the devicetree.h level if necessary IMO.